performancecopilot/pcp
# qa Changes committed to git@github.com:kmcdonell/pcp.git 20191103

Ken McDonell (3):
      qa/common.check: dodge valgind botch on Arch Linux (vm34)
      qa: pm{pmie,logger}_{check,daily} access issues
      qa/common.bcc: openSUSE 15.1 has same bad libbcc as 15.0

 qa/151          |    8 ++++++-
 qa/280          |   12 ++++++-----
 qa/314          |   10 +++++----
 qa/338          |   61 +++++++++++++++++++++++++++++++++-----------------------
 qa/338.out      |   40 ++++++++++++------------------------
 qa/430          |    9 ++++++--
 qa/503          |   52 +++++++++++++++++++++++++++--------------------
 qa/530          |   48 ++++++++++++++++++++++++--------------------
 qa/530.out      |   40 ++++++++++++------------------------
 qa/532          |    8 ++++++-
 qa/641          |    7 +++++-
 qa/658          |    7 +++++-
 qa/664          |    7 +++++-
 qa/925          |    2 -
 qa/common.bcc   |    3 +-
 qa/common.check |    5 +++-
 16 files changed, 181 insertions(+), 138 deletions(-)

Details ...

commit 8b85f9e9560254a88f130b00cf727569b80f5c03
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Mon Nov 4 14:01:56 2019 +1100

    qa/common.bcc: openSUSE 15.1 has same bad libbcc as 15.0
    
    Update pattern to _notrun some tests on this platform.

commit 413e17d23274dc60a10148c321ca0105659b2a16
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Sun Nov 3 20:05:59 2019 +1100

    qa: pm{pmie,logger}_{check,daily} access issues
    
    On Arch Linux (vm34), many tests were failing because these scripts
    do something like this ...
        touch "$PROGLOG"
        chown $PCP_USER:$PCP_GROUP "$PROGLOG" >/dev/null 2>&1
        exec 1>"$PROGLOG" 2>&1
    which does not work with -l $tmp.log from the qa scripts (permission
    issue trying to create $PROGLOG).
    
    Fix is to take better control of permission issues in qa tests and make
    sure everything we need is owned by user $PCP_USER and group $PCP_GROUP.

commit 2c9643a7ef5a6e57aa485ff29e5dd4e10e68defb
Author: Ken McDonell <kenj@kenj.id.au>
Date:   Sun Nov 3 20:03:57 2019 +1100

    qa/common.check: dodge valgind botch on Arch Linux (vm34)
    
    was seeing lines like this ...
        warning: addVar: in range ... outside all rx mapped areas
    ... filter 'em away.
